libraries/AP_AHRS/AP_AHRS_DCM.cpp

```diff
@@ -81,6 +84,13 @@ AP_AHRS_DCM::update(bool skip_ins_update)
     // Perform drift correction
     drift_correction(delta_t);
 
+    if(_ins.get_gyro().z>GYRO_MAX_SPIN_RATE*M_PI/180.0 || _ins.get_gyro().z<-GYRO_MAX_SPIN_RATE*M_PI/180.0)
+    {
+    	//hal.console->printf("pitch, yaw:%f	%f\n\n", _ins.get_pitch_angle_FT()*180.0/3.14, _ins.get_yaw_angle_FT()*180.0/3.14);
+    	
+    	_dcm_matrix.from_euler(0.0f, _ins.get_pitch_angle_FT(), _ins.get_yaw_angle_FT());
```

Comment:
would there be a hard jump at this 180deg point? Would it make sense to interpolate between the values from, lets say, 160deg/s to 200deg/s so that there's never a hard switch? As something ramps up in speed it transitions. Step inputs aren't always a good thing :)